src/librustc_mir/borrow_check/nll/region_infer/values.rs

```diff
@@ -184,18 +184,18 @@ impl ToElementIndex for RegionElementIndex {
 }
 
 /// Stores the values for a set of regions. These are stored in a
-/// compact `BitMatrix` representation, with one row per region
+/// compact `SparseBitMatrix` representation, with one row per region
 /// variable. The columns consist of either universal regions or
 /// points in the CFG.
 #[derive(Clone)]
 pub(super) struct RegionValues {
     elements: Rc<RegionValueElements>,
-    matrix: BitMatrix,
+    matrix: SparseBitMatrix<usize>,
```

Comment:
then this would be `SparseBitMatrix<RegionVid, RegionElementIndex>` -- and some casting below would go away